src/kademlia/node.cpp

```diff
@@ -966,8 +966,12 @@ void node::incoming_request(msg const& m, entry& e)
 		};
 
 		// attempt to parse the message
+		// also reject the message if it has any non-fatal encoding errors
+		// because put messages contain a signed value they must have correct bencoding
+		// otherwise re-encoding the value will break the signature
```

Comment:
It feels a bit wasteful to do a second pass over the buffer to detect soft errors. Could these issues be discovered during the main parsing pass?
I think referring to the problem as "not round-tripping" helps a reader understand it quicker too